java/cuvs-java/src/test/java/com/nvidia/cuvs/DatasetHelper.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.cuvs;
+
+import com.nvidia.cuvs.spi.CuVSProvider;
+import java.lang.foreign.MemorySegment;
+
+public class DatasetHelper {
+
+  public static Dataset fromMemorySegment(MemorySegment memorySegment, int size, int dimensions) {
+    try {
+      return (Dataset)
+          CuVSProvider.provider()
+              .newNativeDatasetBuilder()
```

Comment:
I wonder if we should store the MH directly here in a static final field? (rather than access through the provider each invocation). I'm not sure if it'll help performance or not, but we should try to replicate similar to what we expect the usage to be, no?